src/main/java/it/cnr/isti/hpc/wikipedia/parser/ArticleParser.java

```diff
@@ -538,4 +526,116 @@ private void setDisambiguation(Article a) {
 		}
 	}
 
+	/*
+	 * returns a pair <Namespace, TopicId>
+	 */
+	public static org.apache.commons.math3.util.Pair<String, String> extractNETopic(String target)
+	{
+		int pos = target.indexOf(':');
+		if (pos == -1)
+		{
+			// Doesnt have any NE
+			// i.e: Michael Jackson
+			return new org.apache.commons.math3.util.Pair<String,String>(null, target);
+		}
+		else
+		{
+			// with NE i.e:
+			//  cite:Michael Jackson -> ne: cite, topic: Michael Jacson
+			// ::cite:Michael Jackson -> ne: cite, topic: Michael Jacson
+			// :en:h20:A -> ne: en, Topic: h20:A
+			Pattern patternNE = Pattern.compile(":*([^:]+):(.+)");
+			Matcher m = patternNE.matcher(target);
+			if(m.find()){
+				String ne = m.group(1);
+				String topic = m.group(2);
+				return new org.apache.commons.math3.util.Pair<String,String>(ne, topic);
+
+			}else{
+				// With a colon but without NE
+				// i.e: :Michael Jackson
+				// i.e: ::::Michael Jackson
+				Pattern patternNoNameSpace = Pattern.compile(":*([^:]+.*)");
```

Comment:
[Compile regexes once](https://stackoverflow.com/questions/19829892/java-regular-expressions-performance-and-alternative/19829983#19829983) if at all possible (improves performance).